src/main/java/io/reactivex/Single.java

```diff
@@ -1279,6 +1279,8 @@
      * value and calls a zipper function with an array of these values to return a result
      * to be emitted to downstream.
      * <p>
+     * If the {@code Iterable} of {@link SingleSource}s is empty a {@link NoSuchElementException} error is signalled after subscription.
```

Comment:
I can add explanation if you think it adds value, something like: 

```java
/**
 * Since {@link Single} can't complete without value, there is nothing zip operator
 * can do about zipping 0 {@code Single}s.
 */
```

Reply:
This doesn't sound good to me. Simply state the fact that empty sequence results in a NoSuchElementException.